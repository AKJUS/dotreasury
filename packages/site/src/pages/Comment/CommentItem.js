import React from "react";
import styled from "styled-components";
import { Image } from "semantic-ui-react";
import dayjs from "dayjs";

import Text from "../../components/Text";
import TextMinor from "../../components/TextMinor";
import TextDisable from "../../components/TextDisable";
import Markdown from "../../components/Markdown";

const Wrapper = styled.div`
  padding: 32px 32px 16px;
  :hover {
    background: #FBFBFB;
  }
`

const HeaderWrapper = styled.div`
  display: flex;
  align-items: center;
  margin-bottom: 8px;
`

const Avatar = styled(Image)`
  margin-right: 8px;
`

const Username = styled(TextMinor)`
  margin-right: 16px;
`

const Index = styled(TextMinor)`
  background: #FBFBFB;
  border-radius: 10px;
  height: 20px;
  padding: 0 8px;
  line-height: 20px;
  margin-left: auto;
`

const ContnetWrapper = styled.div`
  margin-left: 32px;
`

const ButtonList = styled.div`
  display: flex;
  align-items: center;
`

const Button = styled.div`
  cursor: pointer;
  opacity: 0.24;
  :not(:last-child) {
    margin-right: 8px;
  }
  :hover {
    opacity: 0.64;
  }
`

const ReplayButton = styled(Button)`
  display: flex;
  align-items: center;
  margin-right: 16px !important;
  & > :first-child {
    margin-right: 5px;
  }
`

const CommentItem = ({comment, index}) => {
  return (<Wrapper>
    <HeaderWrapper>
      <Avatar src="/imgs/avatar.png" />
      <Username>{comment.authorId}</Username>
      <TextDisable>{dayjs(comment.createdAt).format(
        "YYYY-MM-DD HH:mm:ss"
      )}</TextDisable>
      <Index>#{index + 1}</Index>
    </HeaderWrapper>
    <ContnetWrapper>
      <Markdown md={comment.content} />
      <ButtonList>
        <ReplayButton>
          <Image src="/imgs/reply.svg" />
          <Text>Reply</Text>
        </ReplayButton>
        <Button><Image src="/imgs/thumb-up.svg" /></Button>
        <Button><Image src="/imgs/thumb-down.svg" /></Button>
      </ButtonList>
    </ContnetWrapper>
  </Wrapper>)
}

export default CommentItem;
