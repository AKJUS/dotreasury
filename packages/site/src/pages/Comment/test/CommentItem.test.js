import React from 'react';
import renderer from 'react-test-renderer';
import CommentItem from '../CommentItem';

test('Link changes the class when hovered', () => {
  const component = renderer.create(
    <CommentItem comment={{authorId: "600920dc78f01c6de39a9d5c",
    commentId: "600920f278f01c6de39a9d5e",
    content: "test string",
    createdAt: "2021-01-21T06:36:34.229Z",
    updatedAt: "2021-01-21T06:36:34.229Z"}} index={1} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
