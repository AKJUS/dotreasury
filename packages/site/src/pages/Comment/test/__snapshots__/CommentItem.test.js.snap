// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<div
  className="sc-eCssSg cuEDbh"
>
  <div
    className="sc-jSgupP bkzPYG"
  >
    <img
      className="ui image sc-gKsewC eCXYcQ"
      src="/imgs/avatar.png"
    />
    <p
      className="sc-bdfBwQ sc-gsTCUz sc-iBPRYJ inRAOv hiVGgR daatUr"
    >
      600920dc78f01c6de39a9d5c
    </p>
    <p
      className="sc-bdfBwQ sc-dlfnbm inRAOv iVpTwC"
    >
      2021-01-21 14:36:34
    </p>
    <p
      className="sc-bdfBwQ sc-gsTCUz sc-fubCfw inRAOv hiVGgR hNEeEJ"
    >
      #
      2
    </p>
  </div>
  <div
    className="sc-pFZIQ wgauA"
  >
    <div
      className="sc-hKgILt hYsYlF"
    >
      <div
        className="mde-preview-content"
      >
        <p>
          test string
        </p>
      </div>
    </div>
    <div
      className="sc-jrAGrp bykmwW"
    >
      <div
        className="sc-kEjbxe sc-iqHYGH dsjNsK fSCLBs"
      >
        <img
          className="ui image"
          src="/imgs/reply.svg"
        />
        <p
          className="sc-bdfBwQ inRAOv"
        >
          Reply
        </p>
      </div>
      <div
        className="sc-kEjbxe dsjNsK"
      >
        <img
          className="ui image"
          src="/imgs/thumb-up.svg"
        />
      </div>
      <div
        className="sc-kEjbxe dsjNsK"
      >
        <img
          className="ui image"
          src="/imgs/thumb-down.svg"
        />
      </div>
    </div>
  </div>
</div>
`;
